examples/extended_validation/src/tests.rs

```diff
@@ -0,0 +1,89 @@
+use rocket::testing::MockRequest;
+use rocket::http::Method::*;
+use rocket::http::{ContentType, Status};
+
+use super::rocket;
+
+fn test_login(username: &str,
+              password: &str,
+              age: &str,
+              status: Status,
+              body: Option<&'static str>) {
+    let rocket = rocket();
+    let query = &format!("username={}&password={}&age={}", username, password, age);
+
+    let mut req = MockRequest::new(Post, "/login")
+        .header(ContentType::Form)
+        .body(&query);
+
+    let mut response = req.dispatch_with(&rocket);
+    let body_str = response.body().and_then(|body| body.into_string());
+
+    println!("Checking: {:?}/{:?}/{:?}/{:?}", username, password, age, body_str);
+    assert_eq!(response.status(), status);
+
+    if let Some(string) = body {
+        assert!(body_str.map_or(true, |s| s.contains(string)));
+    }
+}
+
+const OK: Status = self::Status::Ok;
+
+#[test]
+fn test_good_login() {
+    test_login("Sergio", "password", "30", Status::SeeOther, None);
+}
+
+#[test]
+fn test_invalid_user() {
+    test_login("-1", "password", "30", OK, Some("Unrecognized user"));
+    test_login("Mike", "password", "30", OK, Some("Unrecognized user"));
+}
+
+#[test]
+fn test_invalid_password() {
+    test_login("Sergio", "password1", "30", OK, Some("Wrong password!"));
+    test_login("Sergio", "ok", "30", OK, Some("Password is invalid: Too short!"));
+}
+
+#[test]
+fn test_invalid_age() {
+    test_login("Sergio",
```

Comment:
Play around with the `test_login` signature and see if we can get these to fit in one line.